File: exercicios/java/dsoex04/src/main/java/org/example/Biblioteca.java
package org.example;

import java.util.ArrayList;
import java.util.List;

public class Biblioteca {
    private List<Livro> livros;

    public Biblioteca() {
        this.livros = new ArrayList<>();
    }

    public List<Livro> getLivros() {
        return this.livros;
    }

    public void incluirLivros(Livro livro) {
        if (livro != null && !this.livros.contains(livro)) {
            this.livros.add(livro);
        }
    }

    public void excluirLivros(Livro livro) {
        if (livro != null) {
            this.livros.remove(livro);
        }
    }
}

File: exercicios/python/Exercicio 04 - Relações entre Objetos (01_09)/biblioteca.py
from livro import Livro

class Biblioteca:
    def __init__(self):
        self.__livros = []

    def incluir_livro(self, livro: Livro):
        if isinstance(livro,Livro):
            for possivel_livro in self.__livros:
                if possivel_livro == livro:
                    return
            self.__livros.append(livro)

    def excluir_livro(self, livro: Livro):
        self.__livros.remove(livro)

    @property
    def livros(self):
        return self.__livros
